public/app/features/alerting/unified/components/rules/central-state-history/CentralAlertHistoryScene.tsx

```diff
@@ -11,35 +11,64 @@ import {
   SceneReactObject,
   SceneRefreshPicker,
   SceneTimePicker,
+  SceneTimeRange,
+  SceneVariableSet,
+  TextBoxVariable,
+  VariableValueSelectors,
+  useUrlSync,
 } from '@grafana/scenes';
+import { GraphDrawStyle, VisibilityMode } from '@grafana/schema/dist/esm/index';
 import {
-  GraphDrawStyle,
   GraphGradientMode,
+  Icon,
   LegendDisplayMode,
   LineInterpolation,
   ScaleDistribution,
   StackingMode,
+  Tooltip,
   TooltipDisplayMode,
-  VisibilityMode,
-} from '@grafana/schema/dist/esm/index';
+  useStyles2,
+} from '@grafana/ui';
+import { Trans } from 'app/core/internationalization';
 
-import { DataSourceInformation, PANEL_STYLES } from '../../../home/Insights';
-import { SectionSubheader } from '../../../insights/SectionSubheader';
+import { DataSourceInformation } from '../../../home/Insights';
 
-import { HistoryEventsListObjectRenderer } from './CentralAlertHistory';
+import { alertStateHistoryDatasource, useRegisterHistoryRuntimeDataSource } from './CentralHistoryRuntimeDataSource';
+import { HistoryEventsListObject } from './EventListSceneObject';
+
```

Comment:
Changes in this file comprises:

- use a new runtime data source for events bar chart (first scence)
- use a new `TextBoxVariable` for sharing the filter by labels.
- sync  scences variables (time range and filter) with the url query params